lib/ModelRoot.js

```diff
@@ -28,11 +28,14 @@ function ModelRoot(o) {
 ModelRoot.prototype.errorSelector = function errorSelector(x, y) {
     return y;
 };
-ModelRoot.prototype.comparator = function comparator(a, b) {
-    if (hasOwn(a, "value") && hasOwn(b, "value")) {
-        return a.value === b.value;
+ModelRoot.prototype.comparator = function comparator(cacheNode, messageNode) {
+    if (hasOwn(cacheNode, "value") && hasOwn(messageNode, "value")) {
```

Comment:
Makes sense, but in that case, isn't there the potential to have other meta-data which impacts the  'distinct' comparison.

I guess what I'm trying to get at is, should we compare all meta-data ($) props, generically, by just iterating them (performance would be a reason not to), instead of picking out expires specifically. I could pass different $foo values in the meta-data arguments in your example above, which would make the nodes distinct.